src/main/java/za/co/dearx/leave/service/LeaveApplicationService.java

```
@@ -81,10 +87,17 @@ public LeaveApplicationDTO create(LeaveApplicationDTO leaveApplicationDTO) throw
             }
         }
 
-        leaveApplication = leaveApplicationRepository.save(leaveApplication);
+        // Use SecurityUtils find the current user's username then looking up a Staff memeber using the username
+        // Defaults to the admin user since it's currently the only user that will not be a staff memeber
```

Nope. I do not agree with this. It is better to throw an exception in this case. But the problem is a bit more difficult.

We will allow an admin user to create a leave application on behalf of someone else. We will have to add conditional logic to the front-end to allow that.

So what I want you to improve on here is this; allow instances where the logged in user has the ADMIN role by using SecurityUtils.hasCurrentUserThisAuthority(). Otherwise make sure we ignore the Staff member in the DTO.

Check if you can get the getCurrentUserLogin(). If not, throw the NotFoundException in the service.exception package. Set an appropriate message.

If you cannot find a staff member for the user, then throw another NotFoundException, stating that there is no staff member linked to the current user.

Then continue as you have done. So you will only enter the block of code to set the Staff member if you could look up a valid value.